pkg/gateway/api/api_grpc.go

```diff
@@ -1004,16 +1192,20 @@ func (s *grpcGatewayService) getSegmentUsers(
 		)
 		return nil, ErrInternal
 	}
-	return res.Users, nil
+	return &featureproto.SegmentUsers{
+		SegmentId: segmentID,
+		Users:     res.Users,
+		UpdatedAt: respGet.Segment.UpdatedAt,
+	}, nil
 }
 
 func (s *grpcGatewayService) getSegmentUsersFromCache(
 	segmentID, environmentId string,
-) ([]*featureproto.SegmentUser, error) {
-	segment, err := s.segmentUsersCache.Get(segmentID, environmentId)
+) (*featureproto.SegmentUsers, error) {
```

Comment:
I refactored to return  `*featureproto.SegmentUsers`, so it can also be used in the `GetSegmentUsers` API.